ideascube/widgets.py

```diff
@@ -1,6 +1,101 @@
 from django.forms import widgets
 
 from ideascube.configuration import get_config
+from django.utils.html import mark_safe, escape
+from django.utils.encoding import force_text
+from django.forms.utils import flatatt
+
+import bleach
+
+
+_script_template = '''<script type="text/javascript">tinymce.init({{
+selector: "#{widget_id}",
+inline: true,
+theme: "inlite",
+hidden_input: false,
+menubar: false,
+toolbar: false,
+selection_toolbar: {settings[selection_toolbar]},
+insert_toolbar: {settings[insert_toolbar]},
+plugins: {settings[plugins]}{settings[extra_settings]}
+}});</script>'''
+
+
+class RichTextEntry(widgets.Widget):
+    def __init__(self, *,
+            with_media = False,
+            selection_toolbar=None,
+            attrs=None):
+        super().__init__(attrs=attrs)
+
+        self.authorized_tags = ['p', 'a', 'ul', 'ol', 'li', 'br',
+                                'h2', 'h3', 'h4', 'h5']
+
+        self.authorized_attributes = {
+            'a': ['href', 'title']
+        }
+
+        if not selection_toolbar is False:
+            selection_toolbar = 'false'
+        else:
+            if selection_toolbar in (True, None):
+                selection_toolbar = '"bold italic | quicklink h2 h3 blockquote"'
+            else:
+                selection_toolbar = '"{}"'.format(' '.join(selection_toolbar))
+
+        insert_toolbar = ['bullist', 'numlist']
+        extra_settings = ['']
+        plugins = ['autolink', 'textpattern']
+        if with_media:
+            extra_settings += ['media_url_resolver: ID.image_url_resolver']
+            insert_toolbar += ['media']
+            plugins += ['media']
+            self.authorized_tags += ['img', 'iframe']
+            self.authorized_attributes['img'] = ['src', 'width', 'height',
+                                                 'alt']
+            self.authorized_attributes['iframe'] = ['src', 'width', 'height',
+                                                    'allowfullscreen']
+
+        self.settings = {
+            'selection_toolbar': selection_toolbar,
+            'insert_toolbar': '"{}"'.format(' '.join(insert_toolbar)),
+            'plugins': '"{}"'.format(' '.join(plugins)) ,
+            'extra_settings': ',\n'.join(extra_settings)
+        }
+
+    def render(self, name, value, attrs=None):
+        if value is None:
+            value = ''
```

Comment:
For what I've saw, the value is None if the form not tied to a existing instance.